### ISSUE TITLE
Support multiple cargo configuration files in `.cargo`

### ISSUE THREAD

**Issue by jonhoo** [rust-lang/cargo#9306]
**Describe the problem you are trying to solve**
I want to check in some subset of my cargo configuration into version control (e.g., [aliases useful to all devs](https://twitter.com/sunshowers6/status/1375256770017304576)), but not others (e.g., parts that hold custom patches, rustflags, or other local build options).

**Describe the solution you'd like**
I would like to be able to split my configuration file into multiple files, and then only check in some of them.

**Notes**
Currently, my options are:
 - Check in the entire cargo configuration into version control, and thus impose my local build configuration on other devs.
 - Don't check in the cargo configuration and find an out-of-band mechanism for distributing common project configuration.
 - Nest my project in an extra subdirectory and place the local configuration in a parent directory and the shared configuration in the project directory itself.

None of these are ideal. It'd be nice if cargo supported something like a `.cargo/config.toml.d/` directory that could then contain any number of configuration files that are merged.

**Comment by Revantus** [comment]
Hi @jonhoo, does splitting your config into `$HOME/.cargo/config.toml` for personal configurations and `/projects/foo/.cargo/config.toml` for project specific, shared configurations not work in this case?
[Cargo Doc Reference](https://doc.rust-lang.org/stable/cargo/reference/config.html#hierarchical-structure)

**Comment by jonhoo** [comment]
No, unfortunately not. It's not uncommon to have personal per-project configuration options (like sharing a `target` directory across related packages or sharing a `rustflags` definition).

**Comment by Revantus** [comment]
I can take a look at implementing this. Using method in [system.conf.d](https://www.freedesktop.org/software/systemd/man/system.conf.d.html) for priority loading based on reverse lexicographical sort. So that `99-personal.toml` would overwrite `00-shared.toml`. First pass implementation would be the single directory config storage and ignore folders inside the `config.toml.d`.

**Comment by ensc** [comment]
perhaps something like an `include` statement could be implemented.  So, `.cargo/config.toml` could look like

```toml
[files]
include = "conf.d/*.toml"
include = "required.toml"
include = "-local.toml"
```

When wildcards expand to null or when prefixed by `-`, files are optional (can be missing).  Else, they are required.

The optional `local.toml` file above can contain e.g. local setup (like `build.target-dir`) and can be excluded e.g. in `.gitignore`.

**Comment by jonhoo** [comment]
There is already an unstable `include` feature that could maybe be extended that way: https://github.com/rust-lang/cargo/issues/7723

**Comment by Revantus** [comment]
That's what I was thinking as well. With the cli taking priority for load order.
